### PR TITLE
fix pledgeId generation

### PR DESCRIPTION
**Pull request by andor0**

### CODE REVIEW

File: src/mapping.ts [hunk 1]
```diff
@@ -518,20 +502,21 @@ export function charge_repay_interest(
...
 // as well as ditribute borrowers locked pTokens among the pledgers
-export function default_pledge_interests(debt: Debt, pBurned: BigInt): void {
+export function default_pledge_interests(debt: Debt, pBurned: BigInt, timestamp: BigInt): void {
   for (let i = 0; i < debt.pledges.length; i++) {
     let pledges = debt.pledges;
     let pledge = get_pledge(pledges[i]);
     let user = get_user(pledge.pledger);
     if (pledge.pledger !== debt.borrower) {
-      let borrower_pledge = get_borrower_pledge(debt);
+      let pledge_hash = pledgeId(pledge.pledger, debt.borrower, debt.id);
```
Comment (in19farkt) [inline src/mapping.ts]:
```diff
- let pledge_hash = pledgeId(pledge.pledger, debt.borrower, debt.id);
+ let pledge_hash = pledgeId(pledge.pledger, debt.borrower, debt.proposal_id);
```

File: src/mapping.ts [hunk 2]
```diff
@@ -693,6 +670,17 @@ function debtProposalId(address: String, proposalId: String): String {
   return construct_two_part_id(address, proposalId);
 }
 
+function pledgeIdFromRaw(sender: Address, borrower: Address, proposalId: BigInt): String {
```
Comment (in19farkt) [inline src/mapping.ts]: `sender` to `supporter`

File: src/mapping.ts [hunk 3]
```diff
@@ -693,6 +670,17 @@ function debtProposalId(address: String, proposalId: String): String {
...
+  return pledgeId(sender.toHexString(), borrower.toHexString(), proposalId.toHex())
+}
+
+function pledgeId(sender: String, borrower: String, proposalId: String): String {
```
Comment (in19farkt) [inline src/mapping.ts]: `sender` to `supporter`

File: src/mapping.ts (hunk 1 above)
Comment (andor0) [reply src/mapping.ts]: done

File: src/mapping.ts (hunk 2 above)
Comment (andor0) [reply src/mapping.ts]: done

File: src/mapping.ts (hunk 3 above)
Comment (andor0) [reply src/mapping.ts]: done

**Review by in19farkt — APPROVED** [review]

**Pull request merged**